addon/mixins/route.js

```diff
@@ -216,42 +122,56 @@ const RouteMixin = Mixin.create({
       delete options.storeFindMethod;
     }
 
-    const startingPage = options.startingPage === undefined ? 0 : options.startingPage-1;
-    const perPage      = options.perPage || this.get('_perPage');
-    const modelPath    = options.modelPath || this.get('_modelPath');
+    const currentPage = options.startingPage === undefined ? 0 : options.startingPage-1;
+    const perPage = options.perPage || 25;
+
+    // check if user passed in param w/ infinityModel, else check if defined on the route (for backwards compat), else default
+    const perPageParam = typeOfCheck(options.perPageParam, get(this, 'perPageParam'), 'per_page');
+    const pageParam = typeOfCheck(options.pageParam, get(this, 'pageParam'), 'page');
+    const totalPagesParam = options.totalPagesParam || 'meta.total_pages';
 
     delete options.startingPage;
     delete options.perPage;
-    delete options.modelPath;
-
-    this.setProperties({
-      currentPage: startingPage,
-      _firstPageLoaded: false,
-      _perPage: perPage,
-      _modelPath: modelPath,
-      _extraParams: options
+    delete options.perPageParam;
+    delete options.pageParam;
+    delete options.totalPagesParam;
+
+    const infinityModel = InfinityModel.create({
+      currentPage,
+      perPage,
+      perPageParam,
+      pageParam,
+      totalPagesParam,
+      _infinityModelName: modelName,
+      extraParams: options,
+      content: A()
     });
 
     this._ensureCompatibility();
 
     if (typeof boundParams === 'object') {
-      this.set('_boundParams', boundParams);
+      let params = {};
+      // get off route before setting on infinityModel
+      Object.keys(boundParams).forEach(k => params[k] = get(this, boundParams[k]));
+      set(infinityModel, '_boundParams', params);
     }
```

Comment:
We spoke about this on slack - I'd like to deprecate bound params.  I don't think it's a good idea to make a binding onto the Route.  IMO, this should be done functionally, by manually updating the params used to build the URL in a one-way-data-flow type of way.

Should we do this in a seperate PR?  Technically, as-is, this will break compat, because right now, bound-params are computed on each query.  This will only compute those params at the time of the `model` hook, and not after.